python/lsst/psf/service/backend/_psf_service_backend.py

```diff
@@ -0,0 +1,343 @@
+            PSF should be evaluated.
+        ref : `lsst.daf.butler.DatasetRef`
+            Fully-resolved reference to a dataset (e.g., `calexp`,
+            `deepCoadd_calexp`, `goodSeeingDiff_differenceExp`).
+
+        Returns
+        -------
+        uri : `lsst.resources.ResourcePath`
+            Full path to the extracted PSF image file.
+        """
+        psf_result = self.extract_ref(ra, dec, ref)
+        return self.write_fits(psf_result)
+
+    def process_uuid(
+        self,
+        ra: float,
+        dec: float,
+        uuid: UUID,
+    ) -> ResourcePath:
+        """Retrieve and write a PSF image from a dataset identified by its
+        UUID.
+
+        Parameters
+        ----------
+        ra, dec : `float`
+            RA/Dec of the point where the PSF should be evaluated (in degrees).
+        uuid : `uuid.UUID`
+            Unique ID of the dataset (e.g., a `calexp`).
+
+        Returns
+        -------
+        uri : `lsst.resources.ResourcePath`
+            Full path to the extracted PSF image file.
+        """
+        psf_result = self.extract_uuid(ra, dec, uuid)
+        return self.write_fits(psf_result)
+
+    def process_search(
+        self,
+        ra: float,
+        dec: float,
+        dataset_type_name: str,
+        data_id: DataId,
+        collections: Sequence[str],
+    ) -> ResourcePath:
+        """Retrieve and write a PSF image from a dataset identified by
+        (dataset type, data ID, collection).
+
+        Parameters
+        ----------
+        ra, dec : `float`
+            RA/Dec of the point where the PSF should be evaluated (in degrees).
+        dataset_type_name : `str`
+            Name of the butler dataset (e.g. "calexp", "deepCoadd_calexp",
+            "goodSeeingDiff_differenceExp").
+        data_id : `dict` or `lsst.daf.butler.DataCoordinate`
+            Data ID used to find the dataset (e.g. {"visit": 12345,
+            "detector": 42}).
+        collections : `collections.abc.Sequence[str]`
+            Collections to search for the dataset.
+
+        Returns
+        -------
+        uri : `lsst.resources.ResourcePath`
+            Full path to the extracted PSF image file.
+        """
+        psf_result = self.extract_search(ra, dec, dataset_type_name, data_id, collections)
+        return self.write_fits(psf_result)
+
+    def extract_ref(
+        self, ra: float, dec: float, ref: DatasetRef, compute_kernel_image: bool = True
+    ) -> PsfExtraction:
+        """Extract a PSF image from a fully-resolved `DatasetRef`.
+
+        Parameters
+        ----------
+        ra, dec : `float`
+            RA/Dec of the point where the PSF should be evaluated (in degrees).
+        ref : `lsst.daf.butler.DatasetRef`
+            Fully-resolved dataset reference.
+        compute_kernel_image : `bool`, optional
+            If `True`, computes the PSF image using `computeKernelImage`;
+            otherwise uses `computeImage`. See Notes for more information.
```

Comment:
We should plumb this new parameter through the other `extract_` methods and `process_` methods so the frontend can use those with the same flexibility.

Reply:
Good catch, done!